Content.Server/Flash/FlashSystem.cs

```diff
@@ -168,6 +170,11 @@ public void FlashArea(Entity<FlashComponent?> source, EntityUid? user, float ran
                 if (!statusEffectsQuery.HasComponent(entity) && !damagedByFlashingQuery.HasComponent(entity))
                     continue;
 
+                if (!_containerSystem.IsInSameOrTransparentContainer((source, null), entity, userSeeInsideSelf: true))
+                {
+                    continue;
+                }
+
                 // Check for entites in view
                 // put damagedByFlashingComponent in the predicate because shadow anomalies block vision.
                 if (!_examine.InRangeUnOccluded(entity, mapPosition, range, predicate: (e) => damagedByFlashingQuery.HasComponent(e)))
```

Comment:
Shouldn't this be covering it?

Reply:
You mean `InRangeUnOccluded` or the predicate itself?

Initially I thought of avoiding adding the container check into `InRangeUnOccluded` itself since I wasn't sure of the other effects. But I did it now, looks to work better. For example, now whispering in another container hides the author (same as it would if occluded in the same container).